### ISSUE TITLE
Preview server fails when using -d switch

### ISSUE THREAD

**Issue by ganondev** [resend/react-email#1043]
### Describe the Bug

Currently, my repository is laid out like this:

```
/
| emails/
  | welcome.tsx
| src/
  | email/
    | templates/
      | welcome.tsx
```

When I do `email dev`, it works great. The catch is that the entire content of `/emails/welcome.tsx` is this:

```ts
import email from '../src/email/templates/welcome';

export default email;
```

This is pretty wasteful. I'd like to be able to just point the cli to where my templates actually are, as it wouldn't make sense for them to be in the default location.

Luckily, I found https://github.com/ganondev/react-email/blob/f4b1ffac888aa5ec558b0179493ae08fbfc61355/packages/react-email/source/index.ts#L18.

I thought I'd try `email dev -d ./src/email/templates`:
![image](https://github.com/resendlabs/react-email/assets/76646757/be31d718-2cfb-4b1f-8b66-bd9d96cfe550)

Console:
```
error - ./src/app/preview/[slug]/page.tsx
ReactServerComponentsError:

You're importing a component that imports react-dom/server. To fix it, render or return the content directly as a Server Component instead for perf and security.

The error was caused by importing '@react-email/render/dist/index.mjs' in './src/app/preview/[slug]/page.tsx'.

Maybe one of these should be marked as a client entry "use client":
  ./src/app/preview/[slug]/page.tsx
```

Not terribly verbose...

### To Reproduce

`email dev -d <...>`

### Expected Behavior

Preview server should just werk

### What's your node version? (if relevant)

18.17.1

**Comment by apetta** [comment]
Same issue here on upgrading from:
`@react-email/components 0.0.7 -> 0.0.12`
`@react-email/render 0.0.7 -> 0.0.12`
`react-email 1.9.5 -> 1.10.0`

**Comment by gabrielmfern** [comment]
This should now be fixed with `react-email@2.0.0`.
